src/main/java/de/hpi/bpt/argos/api/eventType/EventTypeEndpointImpl.java

```diff
@@ -1,93 +1,425 @@
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getEventTypeEntityMappings(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+        JsonArray jsonEntityMappings = new JsonArray();
+
+        try {
+            long eventTypeId = getEventTypeId(request);
+            List<EventEntityMapping> entityMappings = PersistenceAdapterImpl.getInstance().getEventEntityMappingsForEntityType(eventTypeId);
+
+            for (EventEntityMapping entityMapping : entityMappings) {
+                JsonObject jsonEntityMapping = new JsonObject();
+
+                jsonEntityMapping.addProperty("Id", entityMapping.getId());
+                jsonEntityMapping.addProperty("EventTypeId", entityMapping.getEventTypeId());
+                jsonEntityMapping.addProperty("EntityTypeId", entityMapping.getEntityTypeId());
+                jsonEntityMapping.addProperty("TargetStatus", entityMapping.getTargetStatus());
+
+                // add mapping conditions as array
+                JsonArray jsonMappingConditions = new JsonArray();
+                List<MappingCondition> mappingConditions = PersistenceAdapterImpl.getInstance()
+                        .getMappingConditionsForMapping(entityMapping.getId());
+                for (MappingCondition condition : mappingConditions) {
+                    JsonObject jsonCondition = new JsonObject();
+                    jsonCondition.addProperty("EventTypeAttributeId", condition.getEventTypeAttributeId());
+                    jsonCondition.addProperty("EntityTypeAttributeId", condition.getEntityTypeAttributeId());
+                    jsonMappingConditions.add(jsonCondition);
+                }
+                jsonEntityMapping.add("EventEntityMappingConditions", jsonMappingConditions);
+
+                jsonEntityMappings.add(jsonEntityMapping);
+            }
+        } catch (Exception e) {
+            logger.error("cannot parse request body to event entity mappings '" + request.body() + "'");
+            logTrace(e);
+            halt(HttpStatusCodes.ERROR, e.getMessage());
+        }
+        return serializer.toJson(jsonEntityMappings);    }
+
+    /**
+     * This method returns an event type as a JsonObject.
+     * @param eventType - the event type
+     * @return - a json representation of the event type
+     */
+    private JsonObject getEventTypeJson(EventType eventType) {
+        try {
+            JsonObject jsonEventType = new JsonObject();
+
+            jsonEventType.addProperty("Id", eventType.getId());
+            jsonEventType.addProperty("Name", eventType.getName());
+            jsonEventType.addProperty("NumberOfEvents",
+                    PersistenceAdapterImpl.getInstance().getEventCountOfEventType(eventType.getId()));
+            jsonEventType.addProperty("TimestampAttributeId", eventType.getTimeStampAttributeId());
+
+            return jsonEventType;
+        } catch (Exception e) {

```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS1166)